### PR TITLE
feat: add session verification

### PR DESCRIPTION
**Pull request by Abhilasha3112**
*Issue number of the reported bug or feature request: #<number>*

**Describe your changes**
This pr adds session verification using jwt to verify if it is an authorized user accessing homepage. If it is not, it will redirect to login page.

**Testing performed**
Added unit tests. Verified flow on local dev.

### CODE REVIEW

File: src/components/SessionVerification.tsx
Comment (portenez) [inline src/components/SessionVerification.tsx]: Can we address this eslint? I think it wants `response` instead of `res`

File: src/components/SessionVerification.tsx
Comment (portenez) [inline src/components/SessionVerification.tsx]: make this generic:

```typescript
<T extends unknown>(url:string): Promise<T> =>
```

File: src/components/SessionVerification.tsx
Comment (portenez) [inline src/components/SessionVerification.tsx]: return user directly no need for custom extract logic

File: src/pages/home.tsx
Comment (portenez) [inline src/pages/home.tsx]: Hide the path inside `useSessionToken`

**Review by portenez — CHANGES_REQUESTED** [review]

File: src/components/SessionVerification.tsx
Comment (Abhilasha3112) [reply src/components/SessionVerification.tsx]: not sure I understand...if there's an error returned, don't we want to redirect the user to home page?

File: src/pages/home.tsx
Comment (Abhilasha3112) [reply src/pages/home.tsx]: I tried to do this but couldn't..can you please guide me on how to go about it?

File: src/components/SessionVerification.tsx
Comment (Abhilasha3112) [reply src/components/SessionVerification.tsx]: coz if we remove this, we will not be able to distinguish between loading state and error

File: src/pages/_app.tsx
Comment (portenez) [inline src/pages/_app.tsx]:
```typescript
(response) => response.json() as T
````

**Review by portenez — COMMENTED** [review]
One small change to get rid of the type warning.

**Review by portenez — DISMISSED** [review]

**Review by portenez — DISMISSED** [review]
LGTM

**Comment by portenez** [comment]
@Abhilasha3112 , once we got all the approvals, can we squash this PR?

**Review by dakotalillie — DISMISSED** [review]
Looks good!

**Comment by portenez** [comment]
@Abhilasha3122 can you squash? so that we merge? remember coffee chat does
not allow for squash merge.

**Comment by portenez** [comment]
I'm merging this pr, as the last commit it's solely a squash.

**Review by portenez — APPROVED** [review]
The last commit is only a Squash. Verified.

**Pull request merged**
